service/matching/ack_manager.go

```diff
@@ -27,26 +27,42 @@ package matching
 import (
 	"sync"
 
+	"github.com/emirpasic/gods/maps/treemap"
 	"go.uber.org/atomic"
-	"golang.org/x/exp/maps"
 
 	"go.temporal.io/server/common/log"
 	"go.temporal.io/server/common/log/tag"
-	"go.temporal.io/server/common/util"
 )
 
+func int64Comparator(a, b interface{}) int {
```

Comment:
I see there's a `github.com/emirpasic/gods/utils.Int64Comparator`.. why copy it here?

Reply:
Good question. I copied it from the other place in our codebase that uses the `gods` library

Reply:
can we replace them both with that, then?

Reply:
Done.